app/helpers/blacklight_helper.rb

```diff
@@ -0,0 +1,38 @@
+module BlacklightHelper
```

Comment:
I'm not sure this is how we want to expose this behavior out of the arclight engine (for one, if the app has its own `BlacklightHelper`, we won't get this behavior). You might be better off putting the override  in its own helper and having the install generator inject it into the application.

Or, this seems like something that could get ported upstream, perhaps?

Reply:
It is unclear to me what the best way to handle this is.

The logic _could_ be ported upstream to Blacklight. But my understanding is that Blacklight's `page_entries_info` seems to be handing some cases that may not exist anymore. Like this `model_name` stuff? https://github.com/projectblacklight/blacklight/blob/master/app/helpers/blacklight/catalog_helper_behavior.rb#L44-L47

Another hesitation to push this upstream, was where and how we would be able to pass `entry_name` for when the context is a grouped result. ArcLight turns on grouping as a "mode" and Blacklight doesn't really have that understanding as far as I can tell.

Open to suggestions on where we might do this.